marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -535,11 +537,112 @@ function isHubEntityInstance(docUri) {
   return isHubEntityInstance;
 }
 
+function getRecordHistory(docUri) {
+  const history = [];
+  const relations = {'associatedWith': '?', 'attributedTo': '?'};
+  const provenanceRecords = Provenance.findProvenance(docUri, relations);
+
+  if(provenanceRecords.length) {
+    const flows = findFlowsAsMap();
+    const flowNames = getArtifactNamesFromUris(consts.FLOW_COLLECTION, "/flows/", ".flow.json");
```

Comment:
You don't need this anymore, as flowNames = Object.keys(flows).

Also, I recommend renaming "flows" to "flowsMap". That makes it self-documenting that it's a map and not an array or sequence, which is what "flows" implies.